src/main/java/io/jenkins/plugins/folderauth/roles/AbstractRole.java

```diff
@@ -113,7 +118,21 @@ public String getName() {
     @Nonnull
     @SuppressWarnings("unused") // used by index.jelly
     public String getSidsCommaSeparated() {
-        String string = new TreeSet<>(sids).toString();
-        return string.substring(1, string.length() - 1);
+        Jenkins jenkinsInstance = Jenkins.get();
+        SecurityRealm sr = jenkinsInstance.getSecurityRealm();
+        StringBuilder sb = new StringBuilder();
+        for (String sid: new TreeSet<>(sids)) {
+            try {
+                sr.loadUserByUsername(sid);
```

Comment:
For a large number of sids this looks like it can be potentially very slow (if it requires and LDAP lookup). Which IIUC will block the page from loading.

Reply:
+1. Also, this logi will lead to exceptions when handling group membership SIDs

Reply:
I'll look at it again once #58 is done